src/nightly.js

```diff
@@ -60,36 +62,26 @@ module.exports = function*(argv) {
     }
 
     // Clone and update Repos
-    yield repoclone.cloneRepos(repos, /*silent=*/true);
-    yield repoupdate.updateRepos(repos);
+    yield prepareRepos(argv.r);
 
-    //remove local changes and sync up with remote master
-    yield repoutil.forEachRepo(repos, function*() {
-        yield gitutil.gitClean();
-        yield gitutil.resetFromOrigin();
-    });
+    var reposToBuild = flagutil.computeReposFromFlag(argv.r, { includeModules: true });
+    // Get updated nightly versions for all repos
+    /** @type {Object} A map of repo.id and a short SHA for every repo to build */
+    var VERSIONS = yield retrieveVersions(reposToBuild);
 
-    // Get SHAS from repos
-    var SHAJSON = yield retrieveSha(repos);
+    // Update version in package.json and other respective files for every repo
+    // and update dependencies to use nightly versions of packages to be released
+    yield repoutil.forEachRepo(reposToBuild, function*(repo) {
+        apputil.print('Updating ' + repo.id + ' version to ' + VERSIONS[repo.id]);
+        yield versionutil.updateRepoVersion(repo, VERSIONS[repo.id], { commitChanges: false });
 
-    var currentDate = new Date();
-    var nightlyVersion = '-nightly.' + currentDate.getFullYear() + '.' +
-        (currentDate.getMonth() + 1) + '.' + currentDate.getDate();
-    var cordovaLibVersion;
-    //update package.json version for cli + lib, update lib reference for cli
-    yield repoutil.forEachRepo([cordovaLib, cli], function*(repo) {
-        var dir = process.cwd();
-        var packageJSON = require(dir+'/package.json');
-        packageJSON.version = versionutil.removeDev(packageJSON.version) + nightlyVersion +
-            '+' + SHAJSON[repo.id];
-
-        if(repo.id === 'lib'){
-            cordovaLibVersion = packageJSON.version;
-        } else {
-            packageJSON.dependencies['cordova-lib'] = cordovaLibVersion;
-        }
+        var packageJSONPath = path.join(process.cwd(), 'package.json');
```

Comment:
We still need to update dependencies in `package.json` in case if we're releasing both dependent package and its dependency (e.g. `cordova-lib` and `cordova-common`), so the code is still required.

Anyway this code will be updated in the next PR that adds more robust support for platforms nightly releases